packages/cli/src/cmds/validator/handler.ts

```diff
@@ -138,19 +148,40 @@ export async function validatorHandler(args: IValidatorCliArgs & IGlobalArgs): P
   }
 }
 
-function getProposerConfigFromArgs(args: IValidatorCliArgs): ValidatorProposerConfig {
+function getProposerConfigFromArgs(
+  args: IValidatorCliArgs,
+  persistedKeysBackend: IPersistedKeysBackend
+): ValidatorProposerConfig {
+  if (args.flushKeymanagerProposerConfigs) {
+    persistedKeysBackend.deleteProposerConfigs();
+  }
+
   const defaultConfig = {
     graffiti: args.graffiti || getDefaultGraffiti(),
     strictFeeRecipientCheck: args.strictFeeRecipientCheck,
     feeRecipient: args.suggestedFeeRecipient ? parseFeeRecipient(args.suggestedFeeRecipient) : undefined,
     builder: {enabled: args.builder, gasLimit: args.defaultGasLimit},
   };
-  let valProposerConfig;
-  if (args.proposerSettingsFile) {
-    // parseProposerConfig will override the defaults with the arg created defaultConfig
-    valProposerConfig = parseProposerConfig(args.proposerSettingsFile, defaultConfig);
+
+  let valProposerConfig: ValidatorProposerConfig;
+  const proposerConfigFromKeymanager = persistedKeysBackend.readProposerConfigs();
+
+  if (Object.keys(proposerConfigFromKeymanager).length > 0) {
+    // from persistedBackend
+    if (args.proposerSettingsFile) {
+      throw new YargsError(
+        "Cannot accept --proposerSettingsFile since some proposer configs seemed to have been previously persisted via Keymanager api. Use --flushKeymanagerProposerConfigs in conjuction with --proposerSettingsFile if you want to discard them and instead use proposerSettingsFile"
```

Comment:
```suggestion
        "Cannot accept --proposerSettingsFile since it conflicts with proposer configs previously persisted via the keymanager api. Use --flushKeymanagerProposerConfigs in conjuction with --proposerSettingsFile to discard them and instead use the proposerSettingsFile"
```